File: Java/f1xkm2/src/Practical_11/Student.java
package Practical_11;

public class Student implements Comparable<Student> {
    private String student_id;
    private String name;
    private String surname;
    private int age;

    public Student(String student_id, String name, String surname, int age) {
        this.student_id = student_id;
        this.name = name;
        this.surname = surname;
        this.age = age;
    }

    public String getStudent_id() {
        return student_id;
    }

    public String getName() {
        return name;
    }

    public String getSurname() {
        return surname;
    }

    public int getAge() {
        return age;
    }

    public String toString() {
        return "Student ID: " + student_id +
                ", Name: " + name +
                ", Surname: " + surname +
                ", Age: " + age;
    }

    public int compareTo(Student student) {
        return student.student_id.compareTo(this.getStudent_id());
    }


}

File: C++/jasondouble157/Student.cc

#include "Student.h"

Student::Student() {
    this->name = "";
    this->number = "";
}

Student::Student(string name, string number) {
    this->name = name;
    this->number = number;
}

Student::Student(const Student &student) {
    this->name = student.name;
    this->number = student.number;
}

string Student::getName() {
    return this->name;
}

string Student::getNumber() {
    return this->number;
}

bool Student::lessThan(Student &s) {
    if (this->number.compare(s.number) < 0) {
        return true;
    }
    return false;
}


void Student::print() {
    cout << "Name : " << name << "  Number : " << number << endl;
}
